src/utils/apps.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppBinding} from '../types/apps';
+import {AppBindingLocations} from '../constants/apps';
+
+export function fillBindingsInformation(binding?: AppBinding) {
+    if (!binding) {
+        return;
+    }
+
+    binding.bindings?.forEach((b) => {
+        // Propagate id down if not defined
+        if (!b.app_id) {
+            b.app_id = binding.app_id;
+        }
+
+        // Compose location
+        b.location = binding.location + '/' + b.location;
+
+        // Propagate call down if not defined
+        if (!b.call) {
+            b.call = binding.call;
+        }
+
+        fillBindingsInformation(b);
+    });
+
+    // Trim branches without app_id
+    if (!binding.app_id) {
+        binding.bindings = binding.bindings?.filter((v) => v.app_id);
+    }
+
+    // Trim branches without calls
+    if (!binding.call) {
+        binding.bindings = binding.bindings?.filter((v) => v.call);
+    }
+
+    // Pull up app_id if needed
+    if (binding.bindings?.length && !binding.app_id) {
```

Comment:
Also think that we need to document what we do on the user's behalf wrt to their bindings definition.

Reply:
Documented in code, will document also outside.